### ISSUE TITLE
Case sensitive search not supported on index fields

### ISSUE THREAD

**Issue by shubh9194** [vespa-engine/vespa#30476]
**Is your feature request related to a problem? Please describe.**
when we try to search the index field with some value, Vespa returns the result as case insensitive. 
e.g if we are doing search for fieldA="Aabc", it also returns doc that has value "aabc" for fieldA.

**Describe the solution you'd like**
want the case sensitive result only. Docs with fieldsA=Aabc should be returned in the search

**Describe alternatives you've considered**

**Additional context**

**Comment by jobergum** [comment]
Vespa already supports case-sensitive searches using `attribute` fields using `match:cased` [doc](https://docs.vespa.ai/en/reference/schema-reference.html#cased).

 It's not natural to have case sensitive search against `index` field with text matching and linguistic integrations IMHO.

**Comment by bratseth** [comment]
There are use cases for it though, and what we have done recently is to make this completely the discretion of the linguistics module, but the linguistics module will have to support it and most don't. So, I think what's needed is to be able to select a linguistics module per field.

**Comment by 107dipan** [comment]
Hi  @bratseth, 
We are looking to enable  case sensitive search on index fields only with exact match enabled. Is there a way we can achieve case sensitive search on these fields with out using a different lingiustic module since we dont need stemming and other parsing for exact field

**Comment by bratseth** [comment]
To be clear you want both of a) cased exact match, no partial match, b) on an index field not an attribute?

And if so, why can't it be an attribute?

**Comment by 107dipan** [comment]
The only issue with using these fields as attribute is that they are stored in memory or we would need to enable paged similar to our current attributes. Since we already have a lot of attribute with paged enabled this would increase the page in out since these fields are heavily used. 
We can definitely try the attribute approach but were wondering if there is another way.

**Comment by bratseth** [comment]
Ok, got it. No other way right now, but it's not a lot of work to add when also doing "exact" - we can consider doing that.

**Comment by nehajatav** [comment]
Feature request: Allow option to make index fields with {match: word} case sensitive

**Comment by bratseth** [comment]
[match: cased](https://docs.vespa.ai/en/reference/schema-reference.html?#cased) is now supported on index fields.

**Comment by nehajatav** [comment]
Can you share the instructions to upgrade? 
Deployed application with below in schema on 8.466.24, it didnt warn or perform reindexing and it broke searches on field without all lower case values for both existing and new documents - difference in trace query for `fieldName contains "thisString"` before vs after is `fieldName contains {normalizeCase:false}"thisstring"` versus `fieldName contains ("thisString")` 
```
match {
word 
cased
}
```
Upgraded from 8.466.24 to 8.494.23 to 8.503.27, application deploy didnt warn or perform reindexing  and search still broken on above field

**Comment by bratseth** [comment]
Yes, this requires reindexing. Mixed-case tokens will not be recalled until reindexing is complete.

Procedure:
- Deploy the schema change to match cased.
- When self-hosting, reindex, either by refeeding the field in question or by using the reindex API: https://docs.vespa.ai/en/reference/deploy-rest-api-v2.html#reindex (this happens automatically on Vespa Cloud).

We'll add a reindexing message - thanks!

**Comment by nehajatav** [comment]
Reindexing didnt help, mixed case is not working for newer docs either on version 8.503.27

**Comment by bratseth** [comment]
Test with match: word.

**Comment by bratseth** [comment]
Support for the combination with match:word added in 8.506.13.
